meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -182,82 +186,103 @@ public BusinessEntity findByEntityClassAndCode(Class<?> clazz, String code) {
-
-    	fileToDelete.delete();
+    	
+    	File file = GitHelper.getRepositoryDir(currentUser, module.getGitRepository().getCode());
+    	String path = businessEntityDto.getClass().getAnnotation(ModuleItem.class).path() + "/" + entity.getCode();
+    	File removeFile = new File(file, path);
+    	if (removeFile.exists()) {
+    		try {
+				FileUtils.deleteDirectory(removeFile);
+			} catch (IOException e) {
+				throw new BusinessException("Folder unsuccessful deleted : " + removeFile.getPath() + ". " + e.getMessage());
+			}
+    	}
     }
     
     /**
      * Create the entity in the dedicated module
      * 
-     * @param entity
-     * @param module
+     * @param entity belonging to the module
+     * @param module corresponding to the entity
+     * @throws IOException BusinessException
      */
-    public void addFilesToModule(P entity, MeveoModule module) {
-    	MeveoModuleDto meveoModuleDto = new MeveoModuleDto(module);
+    public void addFilesToModule(P entity, MeveoModule module) throws IOException {
     	BusinessEntityDto businessEntityDto = new BusinessEntityDto(entity);
     	String businessEntityDtoSerialize = JacksonUtil.toString(businessEntityDto);
     	
-    	try {
-	    	File file = new File("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
-	    	file.createNewFile();
-    	}catch(IOException e) {
-    		e.printStackTrace();
-    	}
+    	File file = GitHelper.getRepositoryDir(currentUser, module.getGitRepository().getCode());
+    	String path = businessEntityDto.getClass().getAnnotation(ModuleItem.class).path() + "/" + entity.getCode();
     	
-    	try {
-    		Path path = Paths.get("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
-    		byte[] strToBytes = businessEntityDtoSerialize.getBytes();
-
-    		Files.write(path, strToBytes);
-    	}catch(IOException e) {
-    		e.printStackTrace();
-    	}
+    	File newDir = new File (file, path);
+    	
+    	newDir.mkdirs();
+    	
+    	byte[] strToBytes = businessEntityDtoSerialize.getBytes();
+    	
+    	Files.write(newDir.toPath(), strToBytes);
     }
     
     /**
      * Retrieve the current module of the entity, ignore if none
      * Remove the files from old Module
      * Add the files to the new module
      * 
-     * @param entity
-     * @param module
+     * @param entity belonging to the module
+     * @param module corresponding to the entity
+     * @throws BusinessException 
+     * @throws IOException BusinessException
      */
-    public void moveFilesToModule(P entity, MeveoModule module) {
+    public void moveFilesToModule(P entity, MeveoModule module) throws BusinessException {
     	MeveoModule currentModule = findModuleOf(entity);
-    	if (currentModule != null) {
```

Comment:
You misunderstood the comment, the currentModule null check is required there : 

If the entity belongs to a module, remove its files from it.
Then, anyway, add the files to the new module